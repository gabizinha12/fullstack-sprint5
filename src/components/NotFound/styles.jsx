import styled from "styled-components";

export const Image = styled.img`
  height: 100px;
`;

export const Title = styled.h2`
  margin-top: 30px;
  font-size: 20px;
  .link {
    color: #409;
  }
`;
