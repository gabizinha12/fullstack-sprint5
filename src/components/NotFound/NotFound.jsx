import { Title, Image } from "./styles";
import NotFoundImg from "../../assets/not_found.svg";
import { Link } from "react-router-dom";
export function NotFound() {
  return (
    <>
      <Image src={NotFoundImg} alt="Error image" />
      <Title>
        Ops! Parece que a página que você está procurando não existe
        <br />
        <Link to="/" className="link">
          Volte para a página inicial
        </Link>
      </Title>
    </>
  );
}
