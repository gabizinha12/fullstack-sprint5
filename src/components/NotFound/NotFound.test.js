import { render, screen } from "@testing-library/react";
import { NotFound } from "./NotFound";
import { BrowserRouter } from "react-router-dom";

describe("Teste de snapshot not found", () => {
  it("should render not found", () => {
    const { container } = render(
      <BrowserRouter>
        <NotFound />
      </BrowserRouter>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
  // it("should render without errors", () => {
  //   const { container } = render(
  //     <BrowserRouter>
  //       <NotFound />
  //     </BrowserRouter>
  //   );
  //   const element = screen.getByText(
  //     "Ops! Parece que a página que você está procurando não existe"
  //   );
  //   expect(element).toBeInTheDocument();
  // });
});
