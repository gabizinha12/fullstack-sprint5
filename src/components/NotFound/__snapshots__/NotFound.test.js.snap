// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Teste de snapshot not found should render not found 1`] = `
<img
  alt="Error image"
  class="sc-bdnxRM hmQuKM"
  src="not_found.svg"
/>
`;
